### ISSUE TITLE
GetCapacity should allow multiple pools to be returned.

### ISSUE THREAD

**Issue by jieyu** [container-storage-interface/spec#55]
Some CO might prefer install one plugin instance for each storage provider. This greatly simplify the management of external storage providers from CO's perspective.

The current `GetCapacity` RPC in the controller service assumes that there is only one "pool". However, it is very common that a storage provider provides multiple pools of storage, one for each storage grade (aka, class or profile).

When CO creating a volume from a pool, the plugin might expect some plugin specific information about the pool from which the CO wants to create the volume.

```protobuf
message GetCapacityRequest {
  // The API version assumed by the CO. This is a REQUIRED field.
  Version version = 1;
}

message GetCapacityResponse {
  message Result {
    repeated PoolInfo pools = 1;
  }

  // One of the following fields MUST be specified.
  oneof reply {
    Result result = 1;
    Error error = 2;
  }
}

// Describe a storage pool.
message PoolInfo {
  // The total capacity (available + used) of the storage pool from
  // which the controller provisions volumes. This is a REQUIRED
  // field.
  uint64 total_capacity = 1;

  // Plugin specific parameters for the pool as opaque key-value pairs.
  // This field is OPTIONAL. CO SHALL pass these parameters to the
  // Plugin in `CreateVolumeRequest` when creating a volume from this
  // pool.
  map<string, string> parameters = 2;
}
```

**Comment by jdef** [comment]
I'd rather eliminate the `GetCapacity` RPC in favor of a `ListPools` RPC that returns `PoolInfo` objects with optional `total_capacity` and `parameters`

**Comment by cpuguy83** [comment]
I'm not sure the real case for either of these is perfectly solid. I would rather punt on storage pools until after v1 as I don't think it's neccessary for a good plugin.

**Comment by saad-ali** [comment]
I think we can fix this easily by adding the `  map<string, string> parameters` to `GetCapacityRequest`. These would be the same parameters passed to `CreateVolume` and would allow selecting a "pool", e.g. ssd vs sniping disk.

**Comment by msau42** [comment]
The concrete use case for this is for a lvm plugin that manages separate ssd and hdd pools.

**Comment by jieyu** [comment]
I think this issue can be closed since we added `map<string, string> parameters` in `GetCapacityRequest` already. cc @saad-ali @msau42

**Comment by msau42** [comment]
Ah yes this should be sufficient, thanks!

**Comment by jieyu** [comment]
Closed via #85
